[ONNX] Move symbolic export to separate file

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #95935
* #95929
* #95676
* #95664
* #95651
* #95686
* __->__ #95650

Move things around in the effort of preparing to refactor
the code structure.

BowenBao committed:
[ONNX] Move symbolic export to separate file

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move symbolic export to separate file"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Move symbolic export to separate file"


Move things around in the effort of preparing to refactor
the code structure.

[ghstack-poisoned]

wschin commented on torch/onnx/_internal/fx/symbolic_exporter.py:
```diff
@@ -0,0 +1,289 @@
+
+
+@_beartype.beartype
+def _trace_into_fx_graph_via_fx_symbolic_trace(
```
Make it a FX frontend?

BowenBao replied:
Will attempt in a follow up.

wschin commented on torch/onnx/_internal/fx/symbolic_exporter.py:
```diff
@@ -0,0 +1,289 @@
+
+
+@_beartype.beartype
+def _move_placeholder_to_front(graph_module: torch.fx.GraphModule) -> None:
```
There are some graph transforms in this file. They are good candidates for new `Transform` classes.

BowenBao replied:
Done in the pass PR.

BowenBao committed:
Update on "[ONNX] Move symbolic export to separate file"


Move things around in the effort of preparing to refactor
the code structure.

[ghstack-poisoned]
